File: nestjs/07-nest-continent-crud-validation/src/modules/continent/continent.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { ContinentService } from './continent.service';

@Controller('continents')
export class ContinentController {
  constructor(private readonly continentService: ContinentService) {}

  @Get()
  getAll() {
    return this.continentService.getAll();
  }

  @Get(':id')
  getById(@Param('id', ParseIntPipe) id: number) {
    return this.continentService.getById(id);
  }

  @Post()
  create(@Body() body: any) {
    return this.continentService.create(body);
  }

  @Put(':id')
  update(@Param('id', ParseIntPipe) id: number, @Body() body: any) {
    return this.continentService.update(id, body);
  }

  @Delete(':id')
  delete(@Param('id', ParseIntPipe) id: number) {
    return this.continentService.delete(id);
  }
}

File: springboot-java-25/06-springboot-repository-inmemory/src/main/java/com/ganatan/starter/continents/ContinentController.java
package com.ganatan.starter.continents;

import java.util.List;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/continents")
public class ContinentController {

  private final ContinentService service;

  public ContinentController(ContinentService service) {
    this.service = service;
  }

  @GetMapping
  public List<Continent> getAllContinents() {
    return service.getAll();
  }

  @GetMapping("/{id}")
  public Continent getContinentById(@PathVariable int id) {
    return service.getById(id);
  }

  @PostMapping
  @ResponseStatus(HttpStatus.CREATED)
  public Continent createContinent(@Valid @RequestBody ContinentRequestDTO body) {
    return service.create(body.name());
  }

  @PutMapping("/{id}")
  public Continent updateContinent(
    @PathVariable int id,
    @Valid @RequestBody ContinentRequestDTO body
  ) {
    return service.update(id, body.name());
  }

  @DeleteMapping("/{id}")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public void deleteContinent(@PathVariable int id) {
    service.delete(id);
  }
}
